credsweeper/scanner/scan_type/scan_type.py

```diff
@@ -89,43 +90,54 @@ def get_line_data(
+            and filters defined in rule do not remove current line. Empty list - otherwise
 
         """
+        candidates: List[Candidate] = []
         if config.exclude_lines and target.line_strip in config.exclude_lines:
-            return None
-
-        line_data = cls.get_line_data(config=config, target=target, pattern=rule.patterns[0], filters=rule.filters)
-
-        if line_data is None:
-            return None
-        if len(config.exclude_values) > 0 and line_data.value.strip() in config.exclude_values:
-            return None
-
-        candidate = Candidate([line_data], rule.patterns, rule.rule_name, rule.severity, config, rule.validations,
-                              rule.use_ml)
-        # single pattern with multiple values means all the patterns must matched in target
-        if 1 < len(rule.patterns) and rule.rule_type in (RuleType.PATTERN, RuleType.KEYWORD):
-            for pattern in rule.patterns[1:]:
-                aux_line_data = cls.get_line_data(config=config, target=target, pattern=pattern, filters=rule.filters)
-                if aux_line_data is None:
-                    return None
-                if len(config.exclude_values) > 0 and aux_line_data.value.strip() in config.exclude_values:
-                    return None
-                candidate.line_data_list.append(aux_line_data)
-
-        return candidate
+            return candidates
+
+        line_data_list = cls.get_line_data_list(config=config,
+                                                target=target,
+                                                pattern=rule.patterns[0],
+                                                filters=rule.filters)
+
+        for line_data in line_data_list:
+            if len(config.exclude_values) > 0 and line_data.value.strip() in config.exclude_values:
+                continue
+
+            candidate = Candidate([line_data], rule.patterns, rule.rule_name, rule.severity, config, rule.validations,
+                                  rule.use_ml)
+            # single pattern with multiple values means all the patterns must matched in target
+            if 1 < len(rule.patterns) and rule.rule_type in (RuleType.PATTERN, RuleType.KEYWORD):
+                for pattern in rule.patterns[1:]:
+                    aux_line_data_list = cls.get_line_data_list(config=config,
+                                                                target=target,
+                                                                pattern=pattern,
+                                                                filters=rule.filters)
+                    for aux_line_data in aux_line_data_list:
+                        if len(config.exclude_values) > 0 and aux_line_data.value.strip() in config.exclude_values:
+                            continue
+                        candidate.line_data_list.append(aux_line_data)
+                        break
```

Comment:
Why this `for` loop just add only one `aux_line_data`?

Reply:
You are right. Now there are appended at least two line data (left and/or right sides nearest from primary value)